File: CodeWarsJS/2kyuEvaluateMathematicalExpression/CalculatorUtilities.js
"use strict";

module.exports = class CalculatorUtilities{

    static compute(operation, b, a) {
        switch (operation) {
            case "^":
                return Math.Pow(a, b);
            case "*":
                return a * b;
            case "/":
                return a / b;
            case "+":
                return a + b;
            case "-":
                return a - b;
            default:
                return 0;
        }
    }

    static isOperatorOrBracket(item) {
        return this.isOperator(item) || this.isBracket(item);
    }

    static isOperator(item) {
        return item === "^" || item === "*" || item === "/" || item === "+" || item === "-";
    }

    static isBracket(item) {
        return item === "(" || item === ")";
    }

    static isCurrentOperationPriorityBigger(current, old) {
        const  priorityList =
            {
                "^": 1,
                "*": 2,
                "/": 2,
                "+": 3,
                "-": 3,
                "": 4
            };

        if (current === "(" || old === "(") return true;

        return priorityList[old] > priorityList[current];
    }

    static pushStringInCharsToArray(array, string) {
        for (let i = 0; i < string.length; i++) {
            array.push(string[i]);
        }
    }

    static getLastItemInArray(array) {
        return array[array.length - 1];
    }

    static isNextIndexExist(array, currentIndex) {
        return array.length > currentIndex + 1;
    }
}

File: CodeWars/StringCalculator/CalculatorUtilities.cs
using System;
using System.Collections.Generic;

namespace CodeWars.StringCalculator
{
    public static class CalculatorUtilities
    {
        public static double Compute(string operation, double b, double a)
        {
            switch (operation)
            {
                case "^":
                    return Math.Pow(a, b);
                case "*":
                    return a * b;
                case "/":
                    return a / b;
                case "+":
                    return a + b;
                case "-":
                    return a - b;
                default:
                    return 0;
            }
        }

        public static bool IsOperator(string item)
        {
            return item == "^" || item == "*" || item == "/" || item == "+" || item == "-";
        }

        public static bool IsBracket(string item)
        {
            return item == "(" || item == ")";
        }

        public static bool IsOperatorOrBracket(string item)
        {
            return IsOperator(item) || IsBracket(item);
        }

        public static bool IsCurrentOperationPriorityBigger(string current, string old)
        {
            var priorityList = new Dictionary<string, int>
            {
                { "^", 1 },
                { "*", 2 },
                { "/", 2 },
                { "+", 3 },
                { "-", 3 },
                { "", 4 }
            };

            if (current == "(" || old == "(") return true;

            return priorityList[old] > priorityList[current];
        }

    }
}
